dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,234 @@
+package com.dotmarketing.quartz;
+
+import static com.dotmarketing.quartz.DotStatefulJob.getJobDescription;
+import static com.dotmarketing.quartz.DotStatefulJob.getJobGroupName;
+import static com.dotmarketing.quartz.DotStatefulJob.getJobName;
+import static com.dotmarketing.quartz.DotStatefulJob.getTriggerGroupName;
+import static com.dotmarketing.quartz.QuartzUtils.getSequentialScheduler;
+import static java.util.Objects.requireNonNull;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableMap;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalTime;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SimpleTrigger;
+
+public class DotStatefulJobTest extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6152219)